main.go

```diff
@@ -20,20 +21,70 @@ import (
 	"gopkg.in/matryer/try.v1"
 )
 
+const (
+	connectionTimeoutEnv string = "RYUK_CONNECTION_TIMEOUT"
+)
+
 var (
-	port                  = flag.Int("p", 8080, "Port to bind at")
-	initialConnectTimeout = 1 * time.Minute
-	reconnectionTimeout   = 10 * time.Second
+	port                int
+	connectionTimeout   time.Duration
+	reconnectionTimeout time.Duration
 )
 
+type config struct {
+	Port                int
+	ConnectionTimeout   time.Duration
+	ReconnectionTimeout time.Duration
+}
+
+// newConfig parses command line flags and returns a parsed config. config.timeout
+// can be set by environment variable, RYUK_CONNECTION_TIMEOUT. If an error occurs
+// while parsing RYUK_CONNECTION_TIMEOUT the error is returned.
+func newConfig(args []string) (*config, error) {
+	cfg := config{
+		ConnectionTimeout:   60 * time.Second,
+		ReconnectionTimeout: 10 * time.Second,
+	}
+
+	fs := flag.NewFlagSet("ryuk", flag.ExitOnError)
+	fs.SetOutput(os.Stdout)
+
+	fs.IntVar(&cfg.Port, "p", 8080, "Port to bind at")
+
+	err := fs.Parse(args)
+	if err != nil {
+		return nil, err
+	}
+
+	if timeout, ok := os.LookupEnv(connectionTimeoutEnv); ok {
+		parsedTimeout, err := time.ParseDuration(timeout)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse \"%s\": %s", connectionTimeoutEnv, err)
+		}
```

Comment:
Question about this error handling: do we want to fail the execution when the env var value is not a duration or fallback to the default connection timeout?

Reply:
Because we have a default, let's fall back to it if possible.

I think that once we have that, this is ready to go!

Reply:
I think failing is the more Unix-style of doing it. Also, we have to think through that the user for this variable are the Testcontainers implementations. I assume we expect them to always submit a valid duration (unless they would forward user config without any kind of validation). 

So, both approaches have a pro and cons: Either fail more explicitly or be more robust and use a solid fallback. I can live with both 🙂